internal/operands/common-instancetypes/reconcile.go

```diff
@@ -38,72 +50,269 @@ type clusterType interface {
 	instancetypev1alpha2.VirtualMachineClusterInstancetype | instancetypev1alpha2.VirtualMachineClusterPreference
 }
 
-func fetchClusterResources[C clusterType](bundlePath string) ([]C, error) {
-	file, err := ioutil.ReadFile(bundlePath)
-	if err != nil {
-		return nil, err
+func (c *commonInstancetypes) Name() string {
+	return operandName
+}
+
+func WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &instancetypev1alpha2.VirtualMachineClusterInstancetype{}, Crd: instancetypeapi.ClusterPluralResourceName, WatchFullObject: true},
+		{Object: &instancetypev1alpha2.VirtualMachineClusterPreference{}, Crd: instancetypeapi.ClusterPluralPreferenceResourceName, WatchFullObject: true},
+	}
+}
+
+func (c *commonInstancetypes) WatchClusterTypes() []operands.WatchType {
+	return WatchClusterTypes()
+}
+
+func (c *commonInstancetypes) WatchTypes() []operands.WatchType {
+	return nil
+}
+
+func (c *commonInstancetypes) RequiredCrds() []string {
+	return []string{
+		virtualMachineClusterInstancetypeCrd,
+		virtualMachineClusterPreferenceCrd,
+	}
+}
+
+func New(virtualMachineClusterInstancetypeBundlePath, virtualMachineClusterPreferenceBundlePath string) *commonInstancetypes {
+	return &commonInstancetypes{
+		virtualMachineClusterInstancetypeBundle: virtualMachineClusterInstancetypeBundlePath,
+		virtualMachineClusterPreferenceBundle:   virtualMachineClusterPreferenceBundlePath,
 	}
-	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+}
+
+func decodeResources[C clusterType](b []byte) ([]C, error) {
+	decoder := yaml.NewYAMLToJSONDecoder(bytes.NewReader(b))
```

Comment:
This will only decode `b` as yaml, but the function `decodeResMapResources()` calls it with json argument.
Can you leave the json decoder here?

Reply:
https://github.com/kubevirt/ssp-operator/compare/37939ca1995be819c5fb9c0226aec01bd553c396..a5965d7db2de19ef34ce756528e0d4366b62582c is what you are suggesting right @akrejcir ?

Reply:
yes